core/src/components/router/router.tsx

```diff
@@ -118,18 +118,19 @@ export class Router implements ComponentInterface {
   }
 
   /**
-   * Navigate to the specified URL.
+   * Navigate to the specified path.
    *
-   * @param url The url to navigate to.
+   * @param path The path to navigate to.
    * @param direction The direction of the animation. Defaults to `"forward"`.
    */
   @Method()
-  async push(url: string, direction: RouterDirection = 'forward', animation?: AnimationBuilder) {
-    if (url.startsWith('.')) {
-      url = (new URL(url, window.location.href)).pathname;
+  async push(path: string, direction: RouterDirection = 'forward', animation?: AnimationBuilder) {
+    if (path.startsWith('.')) {
+      const currentPath = this.previousPath ?? '/';
+      path = (new URL(path, `https://host/${currentPath}`)).pathname;
```

Comment:
I'm unfamiliar to this syntax. Can you explain a little of what this does/how this works? Also, would this negatively impact/cause issues with developers who are not serving their content through https?

Reply:
URL [second parameter](https://developer.mozilla.org/en-US/docs/Web/API/URL#usage_notes) is a url but we have a path which is only the path in the router a tree.

So basically we are creating a "fake" url just for the purpose of resolving a relative path. To transform a path to a url, you need to add a protocol (`http(s)://`) and a host (`host`).

We only read back the pathname and the search parts so the protocol and host could be anything, this would change the outcome.

Does this make sense ?

Reply:
Perfect, thanks for the explanation that makes a lot of sense! I think it could be helpful to include an inline comment regarding this.

Reply:
Done